test/extended/util/framework.go

```diff
@@ -318,12 +390,21 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface, name stri
 		if err != nil {
 			return err, false
 		}
+		// multiple deployments are conceivable; so we look to see how the latest depoy does
+		var lastRC kapi.ReplicationController
 		for _, rc := range list.Items {
-			err, matched := okOrFailed(&rc)
-			if matched {
-				return err, false
+			// assuming won't have to deal with more than 9 deployments
+			if lastRC.GetName() <= rc.GetName() {
+				lastRC = rc
 			}
 		}
+
+		fmt.Fprintf(g.GinkgoWriter, "\n GGM checking status of rc %s \n", lastRC.GetName())
+		err, matched := okOrFailed(&lastRC)
```

Comment:
lastRC could be uninitialized here since there's no guarantee that list.Items was non-empty.